pkg/network/node/networkpolicy.go

```diff
@@ -674,16 +678,18 @@ func (np *networkPolicyPlugin) watchPods() {
 	np.node.kubeInformers.Core().V1().Pods().Informer().AddEventHandler(funcs)
 }
 
+func isOnPodNetwork(pod *corev1.Pod) bool {
+	if pod.Spec.HostNetwork {
+		return false
+	}
+	return pod.Status.PodIP != ""
+}
+
 func (np *networkPolicyPlugin) handleAddOrUpdatePod(obj, old interface{}, eventType watch.EventType) {
 	pod := obj.(*corev1.Pod)
 	klog.V(5).Infof("Watch %s event for Pod %q", eventType, getPodFullName(pod))
 
-	// Ignore pods with HostNetwork=true, SDN is not involved in this case
-	if pod.Spec.SecurityContext != nil && pod.Spec.HostNetwork {
```

Comment:
(The `pod.Spec.SecurityContext != nil` part here was (harmless) leftover cruft from when the code was ported from `k8s.io/kubernetes/pkg/apis/core` to `k8s.io/api/core/v1`)